array/DNAm/preprocessing/CETYGOdeconvolution.r

```diff
@@ -0,0 +1,195 @@
+
+args<-commandArgs(trailingOnly = TRUE)
+dataDir <- args[1]
+
+gdsFile <-paste0(dataDir, "/2_gds/raw.gds")
+configFile <- paste0(dataDir, "/config.r")
+
+gdsObj<-ifelse(file.exists(gdsFile), TRUE, ifelse(file.exists(msetFile), FALSE, NA))
+
+source(configFile)
+
+
+#----------------------------------------------------------------------#
+# LOAD PACKAGES
+#----------------------------------------------------------------------#
+
+library(CETYGO)
+
+if(is.na(gdsObj)){
+	message("No data to load")
+} else{
+	if(gdsObj){
+		message("Loading data from gds object")
+		library(bigmelon)
+	} else {
+		if(!gdsObj){
+			message("Loading data from mset object")	
+			library(wateRmelon)
+		}
+}
+}
+
+
+
+#----------------------------------------------------------------------#
+# IMPORT DATA
+#----------------------------------------------------------------------#
+
+setwd(dataDir)
+library(gridExtra)
+
+# load sample sheet
+sampleSheet<-read.csv("0_metadata/sampleSheet.csv", na.strings = c("", "NA"), stringsAsFactors = FALSE)
+# if no column Basename, creates from columns Chip.ID and Chip.Location
+if(!"Basename" %in% colnames(sampleSheet)){
+	sampleSheet$Basename<-paste(sampleSheet$Chip.ID, sampleSheet$Chip.Location, sep = "_")
+}
+
+if(gdsObj){
+
+	gfile<-openfn.gds(gdsFile, readonly = FALSE, allow.fork = TRUE)
+	# ensure sample sheet is in same order as data
+	sampleSheet<-sampleSheet[match(colnames(gfile), sampleSheet$Basename),]
+	# extract a few useful matrices
+	if(toupper(arrayType) == "V2"){
+	  rawbetas<-epicv2clean(betas(gfile)[])
+} else {
+	rawbetas<-betas(gfile)[,]
+}
+    closefn.gds(gfile)
+}else {
+	if(!gdsObj){
+		load(msetFile)
+		# ensure sample sheet is in same order as data
+		sampleSheet<-sampleSheet[match(colnames(msetEPIC), sampleSheet$Basename),]
+		# extract a few useful matrices
+		rawbetas<-betas(msetEPIC)
+}
+}
+
+
+  
+
+
+#----------------------------------------------------------------------#
+# DEFINE CETYGO FUNCTION
+#----------------------------------------------------------------------#
+
+adultBrainCETYGO <- function(betas, cellType){
+  
+  predPropAll<-list()
+  CETYGOplots <- list()
+  propPlots <- list()
+  counter = 1
+  maxCETYGO <- 0
+  minCETYGO <- 1
+  
+  for(method in names(modelBrainCoef)){
+    for(j in 1:length(modelBrainCoef[[method]])){
+      if(!is.null(modelBrainCoef[[method]][[j]])){
+        predPropAll[[method]][[j]]<-projectCellTypeWithError(betas, modelBrainCoef[[method]][[j]])
+      }
+    } 
+  }
+  
+  
+  for(i in 1:length(predPropAll)){
+    type <- predPropAll[[1]]
+    for(j in 1:length(type)){
+      plotdf <- as.data.frame(type[[j]])
+      
+      if (ncol(plotdf) > 0) {
+        
+        if (max(plotdf$CETYGO) > maxCETYGO){
+          maxCETYGO <- max(plotdf$CETYGO)}
+        
+        if (min(plotdf$CETYGO) < minCETYGO){
+          minCETYGO <- min(plotdf$CETYGO)}
+      }
+    }
+  }
+  
+  for(i in 1:length(predPropAll)){
+    type <- predPropAll[[1]]
```

Comment:
should this be `type <- predpPropAll[[i]]`?

Reply:
yes. Have changed script